test/test_va_api_fixture.cpp

```diff
@@ -507,6 +638,88 @@ void VAAPIFixture::doCreateConfigToFail(VAProfile profile,
               VA_STATUS_ERROR_INVALID_CONFIG);
 }
 
+void VAAPIFixture::doUploadImage()
+{
+    int box_width_loc=8;
+    int row_shift_loc=0;
+    int i;
+
+    for (i=0; i<SURFACE_NUM; i++) {
+        upload_surface(m_vaDisplay, m_surfaceID[i], box_width_loc, row_shift_loc, 0);
+
+        row_shift_loc++;
+        if (row_shift_loc==(2*box_width_loc)) row_shift_loc= 0;
+    }
+}
+
+void VAAPIFixture::doGetImage(uint32_t currentFmt, VASurfaceID surface_id,
+                                std::pair<uint32_t, uint32_t> currentResolution)
+{
+    doTestCreateImage(currentFmt, currentResolution);
+
+    ASSERT_STATUS(vaGetImage(m_vaDisplay, surface_id, 0, 0,
+                currentResolution.first, currentResolution.second, m_image.image_id));
+}
+
+VASurfaceID VAAPIFixture::doGetNextSurface(int *index)
```

Comment:
make this a reference instead of a raw pointer

Reply:
Reworked.